lib/Runtime/Types/DictionaryTypeHandler.cpp

```diff
@@ -869,6 +869,81 @@ namespace Js
     }
 
     template <typename T>
+    BOOL DictionaryTypeHandlerBase<T>::DeleteProperty(DynamicObject *instance, JavascriptString *propertyNameString, PropertyOperationFlags propertyOperationFlags)
+    {
+        AssertMsg(!PropertyRecord::IsPropertyNameNumeric(propertyNameString->GetString(), propertyNameString->GetLength()),
+            "Numeric property names should have been converted to uint or PropertyRecord* ");
+
+        ScriptContext* scriptContext = instance->GetScriptContext();
+        DictionaryPropertyDescriptor<T>* descriptor;
+        JsUtil::CharacterBuffer<WCHAR> propertyName(propertyNameString->GetString(), propertyNameString->GetLength());
+
+        if (propertyMap->TryGetReference(propertyName, &descriptor))
+        {
+            Assert(descriptor->SanityCheckFixedBits());
+
+            if (descriptor->Attributes & PropertyDeleted)
+            {
+                return true;
```

Comment:
>return true; [](start = 16, length = 12)

If i compare the code from DeleteProperty_Internal() below, there is some code missing here. Is that intentional?
```c++
 // If PropertyDeleted and PropertyLetConstGlobal are set then we have both
                // a deleted global property and let/const variable in this descriptor.
                // If allowLetConstGlobal is true then the let/const shadows the property
                // and we should return false for a failed delete by going into the else
                // if branch below.
                if (allowLetConstGlobal && (descriptor->Attributes & PropertyLetConstGlobal))
                {
                    JavascriptError::ThrowCantDeleteIfStrictMode(propertyOperationFlags, scriptContext, propertyRecord->GetBuffer());

                    return false;
                }
```